dask/array/tests/test_cupy.py

```diff
@@ -0,0 +1,56 @@
+import numpy as np
+import pytest
+
+import dask.array as da
+from dask.array.utils import assert_eq
+
+cupy = pytest.importorskip('cupy')
+
+
+functions = [
+    lambda x: x,
+    pytest.mark.xfail(lambda x: da.expm1(x), reason="cupy doesn't support expm1"),
+    lambda x: 2 * x,
+    lambda x: x / 2,
+    lambda x: x**2,
+    lambda x: x + x,
+    lambda x: x * x,
+    lambda x: x[0],
+    lambda x: x[:, 1],
+    lambda x: x[:1, None, 1:3],
+    lambda x: x.T,
+    lambda x: da.transpose(x, (1, 2, 0)),
+    lambda x: x.sum(),
+    pytest.mark.xfail(lambda x: x.dot(np.arange(x.shape[-1])), reason='cupy.dot(numpy) fails'),
```

Comment:
This was the "weird" ufunc? May be good to leave a comment here with link to issue, if possible.

Reply:
The reason that this fails is that a `my_cupy_array.dot(my_numpy_array)` fails in cupy.  Do you have suggestions on how to improve the text in the reason listed above?